tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -1124,7 +1110,7 @@ private boolean doTruncateInvalidTxBefore(long time) throws InvalidTruncateTimeE
     
     // Find all invalid transactions earlier than truncateWp
     Set<Long> toTruncate = Sets.newHashSet();
```

Comment:
Actually this can be a `LongSet` so that it can be passed to the `doTruncateInvalidTx` and eventually to `invalidTxList.removeAll` as a `LongSet` to have more efficient removal.